evolver/hardware/standard/od_sensor.py

```diff
@@ -0,0 +1,35 @@
+from pydantic import Field
+
+from evolver.connection.interface import Connection
+from evolver.hardware.interface import SensorDriver, VialBaseModel, VialConfigBaseModel
+from evolver.serial import SerialData
+
+
+class ODSensor(SensorDriver):
+    """Optical density sensor driver
+
+    This driver represents a family of sensors for turbidity which are read by
+    integrating a specified number of ADC readings per vial.
+    """
+
+    class Config(VialConfigBaseModel):
+        addr: str = Field(description="Address of od sensor on serial bus (e.g. od_90)")
+        integrations: int = Field(500, description="on read, request average of this number of ADC reads")
+        serial_conn: Connection | None = Field(None, description="serial connection, default is that on evolver")
+
+    class Output(VialBaseModel):
+        raw: int
+        density: float = None
+
+    @property
+    def serial(self):
+        return self.serial_conn or self.evolver.serial
+
+    def read(self):
+        self.outputs = {}
+        cmd = SerialData(addr=self.addr, data=[str(self.integrations).encode()], kind="r")
+        with self.serial as comm:
+            response = comm.communicate(cmd)
+        for vial, raw in enumerate(response.data):
+            if vial in self.vials:
+                self.outputs[vial] = self.Output(vial=vial, raw=raw)
```

Comment:
If ``self.vials`` is updated once live, and previous values are taken and thus ``self.outputs`` is already populated, who's responsibility is it to set/reset ``self.outputs``? Here we only set the ones we think we want, but ``get`` returns ``self.outputs`` which could include stale data. It may be better to address that here.

Reply:
It would be good to add a test for this.

Reply:
Another read would have output only include a new set of vials, I can add a test for this

Reply:
ah, indeed `get` data are sort of by definition stale (its like get access to previous read). Would have to think about what it would mean to update the vial list in place and whether that should affect the gets before next read.

Reply:
>Would have to think about what it would mean to update the vial list in place and whether that should affect the gets before next read.

I agree, but to me, this just highlights a root issue with separating `read`` and ``get`` like this. It doesn't seem intuitive and also requires additional semantics and functionality to make it correct. As also noted above in this PR https://github.com/ssec-jhu/evolver-ng/pull/72#discussion_r1633264022.

Reply:
ok, that is another thing, we can `get` to that later 🤣 ,  I'll leave the behavior as-is for this change.

Reply:
ba dum tshh